### ISSUE TITLE
No way to make legend order exactly match bar order in relative stacked bar plot

### ISSUE THREAD

**Issue by dracodoc** [ropensci/plotly#1694]
Sometimes we want to plot negative values below x axis in a stacked bar plot. [This can be done with barmode as "relative"](https://community.plot.ly/t/stacked-bar-with-positive-and-negative-values/920). 

Further, I want to make the legend order exactly match the bar order in plot. However the positive bars stack from middle to top, negative bars stack from middle to down, thus no matter how I arrange the variable levels, there is no way to make the legend order exactly match bar order. 

```r
library(plotly)
library(data.table)

set.seed(42)
dt_test <- data.table(x = rep(1:5, each = 6), variable = rep(letters[1:6], 5), value = sample(10:40, 30))
dt_test[variable %in% letters[3:5], value := -value]

dt_test[, variable := factor(variable, levels = c("c", "d", "e", "a", "b", "f"))]

plot_ly(dt_test) %>% 
  add_bars(x = ~x, y = ~value, color = ~variable) %>%
  layout(barmode = 'relative', legend = list(traceorder = "reversed"))
```
![image](https://user-images.githubusercontent.com/1730496/73885138-44070a00-4835-11ea-9e65-22f9706ef64e.png)

I'm not sure how this can be done, unless we 
- change the stack order in negative part as bottom up
- or can arrange legend order freely without changing bar stack order

**Comment by WhyBeeYes** [comment]
Yes, you can arrange legend order freely without changing bar stack order.  Then use `scale_fill_manual` to define your own breaks and color as shown below:
```
barorderr <- c("a", "b", "f", "e", "d", "c")
mycolor <- c("red", "blue", "purple", "green", "darkblue", "orange")

p <- ggplot(dt_testt, aes(x=x , y=value, fill=factor(variable) )) + 
  scale_fill_manual(name = "my variable", 
                    breaks = barorderr, 
                    values = mycolor) +
  geom_bar(stat = "identity", alpha=0.5)
```
![image](https://user-images.githubusercontent.com/63007756/78506558-58d62000-7748-11ea-9271-1fc3e70dd039.png)

**Comment by dracodoc** [comment]
I'm not sure if I fully follows your thoughts, because I was referring to using plotly syntax directly, not making a ggplot then convert to plotly with ggplotly call. The conversion works well for simple cases but cannot handle certain details especially with facet, so I tend to use plotly syntax directly when I need more control.
